docs/adr/adr-001-predevnet-celestia-node.md

```diff
@@ -0,0 +1,131 @@
+    * `BlockExchange` (`full` node <> `full` node)
+    * `BlockStore`
+* `StateService` *(optional for devnet)*
+    * `StateExchange`
+* `TransactionService` *(dependent on `StateService` implementation, but optional for devnet)*
+    * `SubmitTx`\
+
+
+For devnet, it should be possible for Celestia `full` Nodes to receive information directly from Celestia Core nodes or from each other. 
+
+## Considerations 
+
+### State Fraud Proofs
+For the Celestia Node to be able to propagate `StateFraudProof`s, we must modify Celestia Core to store blocks with invalid state and serve them to both the Celestia Node and the Celestia App, **and** the Celestia App must be able to generate and serve `StateFraudProof`s via RPC to Celestia nodes.
+
+This feature is not necessarily required for devnet (so state exection functionality for Celestia Full Nodes can be stubbed out), but it would be nice to have for devnet as we will likely allow Celestia Full Nodes to speak with other Celestia Full Nodes instead of running a trusted Celestia Core node simultaenously and relying on it for information.
+
+A roadmap to implementation could look like the following: 
+
+The Celestia Full Node would adhere to the ABCI interface in order to communicate with the Celestia App (similar to the way Optimint does it). The Celestia Full Node would send State requests to the Celestia App in order for the Celestia app to replay the transactions in the block and verify the state. 
+
+For devnet, it is okay to stub out state verification functionality. For example, a Celestia Full Node would download reserve transactions, but not replay them. 
+
+
+### Fraud Proofs
+
+For validators (Celestia Core) to be able to sign blocks, they need to erasure code the block data and commit to the data root of the erasure coded block that is stored in the header. The validators do not need to store the erasure coded block, but only the "raw" block with the data root in the header. A Celestia Node will fetch the "raw" block from Celestia Core via the `/block` endpoint, and then erasure code the block, checking it against the data root in the header, and then store the erasure coded block.
+
+For devnet, the Celestia Node will not be able to generate state fraud proofs as it will not have a state execution environment, so it will rely on Celestia Core to produce state fraud proofs and retrieve the proof via a `/state_fraud` endpoint. The Celestia Node will also require the "state fraudulent" block, so Celestia Core needs to make those blocks available via the `/block` endpoint.
+
+### Light nodes serving shares and samples
+
+At the moment, we will be using [bitswap](https://github.com/ipfs/go-bitswap) to retrieve samples and shares from the network. The way Bitswap works requires nodes that have the requested data to serve it. This is not necessarily ideal for a "light node" to do as supporting serving samples/shares would expand the resource requirements for a light node.
```

Comment:
bitswap has other downsides too:
 - it somewhat couples storage with networking (e.g. it will be difficult to just compute the inner proof nodes on demand instead of storing them)
 - by default, it requires multiple roundtrips for what could be a single (or at least fewer) roundtrip(s) if we wrote our own protocol; this is particularly relevant in the case where Celestia nodes will run on their own and download the whole block via the p2p network from other Celestia nodes (instead of from tendermint via RPC): [the current implementation](https://github.com/celestiaorg/celestia-core/blob/052d1269e0ec1de029e1cf3fc02d2585d7f9df10/p2p/ipld/read.go#L23-L30) using bitswap and ipfs is quite inefficient compared to a protocol that was not bitswap on a share-level
 - more code we do not directly have control over and more dependencies